spock-core/src/main/resources/org/spockframework/idea/spock.gdsl

```diff
@@ -25,7 +25,8 @@ def ctx = context(ctype: "spock.lang.Specification", scope: closureScope(isArg:
 // Foo x = Mock { /* delegates to instance of type Foo */ }
 contributor(ctx, {
   def call = enclosingCall("Mock") ?: enclosingCall("Stub") ?: enclosingCall("Spy") ?:
-    enclosingCall("GroovyMock") ?: enclosingCall("GroovyStub") ?: enclosingCall("GroovySpy")
+          enclosingCall("GroovyMock") ?: enclosingCall("GroovyStub") ?: enclosingCall("GroovySpy") ?:
+                  enclosingCall("MockStatic") ?: enclosingCall("StubStatic") ?: enclosingCall("SpyStatic")
```

Comment:
This doesn't work, the type on the left hand side is `IStaticMock<T>` so the delegation would not show the correct type. You even added a test for `IStaticMock<StaticClass> mock = MockStatic()` not working.